rest/views.py

```diff
@@ -465,3 +482,109 @@ def get_queryset(self):
+    permission_classes = (
+        ViewRepoPermission,
+        IsAuthenticated,
+    )
+
+    def get_queryset(self):
+        repo_slug = self.kwargs['repo_slug']
+        try:
+            exports = self.request.session[
+                EXPORTS_KEY][repo_slug]
+        except KeyError:
+            exports = []
+        return [{"id": lr_id} for lr_id in exports]
+
+    def create(self, request, *args, **kwargs):
+        try:
+            lr_id = int(request.data['id'])
+        except ValueError:
+            raise ValidationError("learning resource id must be a number")
+        repo_slug = self.kwargs['repo_slug']
+
+        learning_resource = get_resource(lr_id, self.request.user.id)
+        if learning_resource.course.repository.slug != repo_slug:
+            raise PermissionDenied()
+
+        if EXPORTS_KEY not in self.request.session:
+            self.request.session[EXPORTS_KEY] = {}
+
+        if repo_slug not in self.request.session[EXPORTS_KEY]:
+            self.request.session[EXPORTS_KEY][repo_slug] = []
+
+        exports = self.request.session[EXPORTS_KEY][repo_slug]
+        if lr_id not in exports:
+            exports.append(lr_id)
+            self.request.session.modified = True
+
+        url = reverse('learning-resource-export-detail',
+                      kwargs={'repo_slug': self.kwargs['repo_slug'],
+                              'lr_id': lr_id})
+        headers = {"Location": url}
+        return Response(
+            {"id": lr_id}, status=status.HTTP_201_CREATED, headers=headers)
+
+    # pylint: disable=unused-argument
+    def delete(self, request, *args, **kwargs):
+        """Delete all ids in session for this repository and user"""
+        try:
+            repo_slug = self.kwargs['repo_slug']
+            del self.request.session[EXPORTS_KEY][repo_slug]
+            self.request.session.modified = True
+        except KeyError:
+            # doesn't exist, no need to delete
+            pass
+        return Response(
+            status=status.HTTP_204_NO_CONTENT,
+            content_type="text"
+        )
+
+
+class LearningResourceExportDetail(RetrieveDestroyAPIView):
+    """
+    Detail resource for a learning resource id for export
+    """
+    serializer_class = LearningResourceExportSerializer
+    permission_classes = (
+        ViewLearningResourcePermission,
+        IsAuthenticated,
+    )
+    lookup_field = 'id'
+    lookup_url_kwarg = 'lr_id'
+
+    def get_object(self):
+        try:
+            lr_id = int(self.kwargs['lr_id'])
+        except ValueError:
+            raise Http404
+
+        repo_slug = self.kwargs['repo_slug']
+        try:
+            exports = self.request.session[
+                EXPORTS_KEY][repo_slug]
+        except KeyError:
+            raise Http404
+
+        if lr_id not in exports:
+            raise Http404
+
+        return {"id": lr_id}
+
+    def perform_destroy(self, instance):
+        lr_id = instance['id']
```

Comment:
Pylint doesn't require it since it overrides a method but in this case it's probably a good idea anyway